### PR TITLE
Mrbean/add disable shape inference

### PR DESCRIPTION
**Pull request by sam-h-bean**
# What does this PR do?
Allows to disable shape inference in quantization. This is the fix suggested in [this issue](https://github.com/microsoft/onnxruntime/issues/11072) but there was currently no way to leverage this configuration in Optimum.

Fixes # (issue)
https://github.com/huggingface/optimum/issues/300

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sam-h-bean** [comment]
@lewtun @michaelbenayoun @JingyaHuang  what do you think?

**Comment by sam-h-bean** [comment]
pinging @echarlaix and @JingyaHuang once again. This is a blocker for quantizing very large models so would love to see this go in!

**Comment by sam-h-bean** [comment]
@lewtun I saw there were some tests around static quantization in `tests/test_optimization.py` so I put a unit test in there. That work for you?

File: .gitignore [hunk 1]
```diff
@@ -134,4 +134,5 @@ dmypy.json
 # include small test model for tests
 !tests/assets/onnx/model.onnx
 
-.vscode
\ No newline at end of file
+.vscode
+.idea
```
Comment (JingyaHuang) [inline .gitignore:138]:
```suggestion
```
Can you remove the change on `.gitignore`? As it is irrelevant to the issue.

File: optimum/onnxruntime/configuration.py [hunk 2]
```diff
@@ -277,10 +277,12 @@ class QuantizationConfig:
     qdq_op_type_per_channel_support_to_axis: Dict[str, int] = field(
         default_factory=lambda: ORT_DEFAULT_CHANNEL_FOR_OPERATORS
     )
+    disable_shape_inference: bool = False
```
Comment (JingyaHuang) [inline optimum/onnxruntime/configuration.py:280]: Can you also add a description of the attribute in the docstring? Thx!

File: tests/onnxruntime/test_optimization.py [hunk 3]
```diff
@@ -234,6 +234,10 @@ def preprocess_function(examples, tokenizer):
             self.assertEqual(expected_quantized_matmul, num_quantized_matmul)
             gc.collect()
 
+    def test_static_quantization_without_shape_inference_raises(self):
+        with self.assertRaises(expected_exception=ValueError):
+            QuantizationConfig(is_static=True, disable_shape_inference=True, format=QuantFormat.QDQ)
```
Comment (JingyaHuang) [inline tests/onnxruntime/test_optimization.py:239]: Seems to be the correct place to put this if we insist on adding a test for `__pos_init__`, pinging @echarlaix for confirmation.

**Review by JingyaHuang — APPROVED** [review]
LGTM, thanks for taking care of it, and sorry for the delay. Just left several small nits, and can you rebase it with the main branch?

File: .gitignore (hunk 1 above)
Comment (sam-h-bean) [reply .gitignore:138]: Can we just leave it? Do we really need a dedicated PR for adding .idea to the .gitignore?

File: optimum/onnxruntime/configuration.py (hunk 2 above)
Comment (sam-h-bean) [reply optimum/onnxruntime/configuration.py:282]: done!

File: tests/onnxruntime/test_optimization.py (hunk 3 above)
Comment (sam-h-bean) [reply tests/onnxruntime/test_optimization.py:239]: you let me know!

**Comment by sam-h-bean** [comment]
Closing this PR as it seems to not be a priority @JingyaHuang @lewtun

**Pull request closed without merging**